hrms/hr/doctype/expense_claim/expense_claim.py

```diff
@@ -55,21 +55,22 @@ def set_status(self, update=False):
 		precision = self.precision("grand_total")
 
 		if (
-			# set as paid
-			self.is_paid
-			or (
-				flt(self.total_sanctioned_amount) > 0
-				and (
-					# grand total is reimbursed
-					(
-						self.docstatus == 1
-						and flt(self.grand_total, precision) == flt(self.total_amount_reimbursed, precision)
+			(
+				# set as paid
+				self.is_paid
+				or (
+					flt(self.total_sanctioned_amount) > 0
+					and (
+						# grand total is reimbursed
+						(flt(self.grand_total, precision) == flt(self.total_amount_reimbursed, precision))
+						# grand total (to be paid) is 0 since linked advances already cover the claimed amount
+						or (flt(self.grand_total, precision) == 0)
 					)
-					# grand total (to be paid) is 0 since linked advances already cover the claimed amount
-					or (flt(self.grand_total, precision) == 0)
 				)
 			)
-		) and self.approval_status == "Approved":
+			and self.docstatus == 1
+			and self.approval_status == "Approved"
```

Comment:
Let's move these easier conditions up so that it immediately moves to `elif` if any of this fails.